### DISCUSSION TITLE
Web follower (Librelinkup) increased reports of 403 error

### DISCUSSION THREAD

**Discussion by PhilTU** [NightscoutFoundation/xDrip#3808] [Bug reports]
More users (Facebook, Reddit) are reporting loss of connectivity with Web follow getting 403 errors when trying to collect data from Librelinkup data source using xDrip+

The vendor has made changes that require a later version of the Linkup app to be identified in headers, and additional parameters to be added. 

#3794 and late additions to #2893 provide screenshots etc.

**Comment by PetBom** [comment]
I have the issue too, and it seems that more users in my network is experiencing the same issue. I can pinpoint the start of the problem to 11:00 CET. No changes, network issues, updates on my client at that time. So something probably changed on the server side.

**Comment by Navid200** [comment]
Have these troubleshooting steps been tried?
https://navid200.github.io/xDrip/docs/Libre3_Follow.html

**Reply by PetBom** [reply]
Yes. I have.

**Reply by PhilTU** [reply]
> Have these troubleshooting steps been tried? https://navid200.github.io/xDrip/docs/Libre3_Follow.html

I am not asking people to do that given the specific error message and (as far as I know) no solution has been implemented yet. Please correct me if I am wrong.

**Reply by davidcharlat** [reply]
It seems that this link is for librelink3 users...
I use librelink2 and i got the same problem

**Reply by duncanScalion** [reply]
yes libre 2 users we have same issue

**Comment by gui-dos** [comment]
A new "Account-Id" HTTP header is now required, passing the SHA256 digest of a user's` id` as a 64-char hexadecimal string.

You can have a look at my Swift one-liners I PRed with https://github.com/JohanDegraeve/xdripswift/commit/13e4a1e7 ;-)

**Reply by PhilTU** [reply]
> A new "Account-Id" HTTP header is now required, passing the SHA256 digest of a user's` id` as a 64-char hexadecimal string.

from what you know would this be in the xDrip code or in the script that lives in the shadows.

**Comment by mariajosepela** [comment]
@gui-dos, podrías explicarnos cómo realizar esos cambios, gracias . No entiendo mucho de informática

**Reply by PhilTU** [reply]
> @gui-dos, could you explain how to make those changes, thank you. I don't understand much about computers

They are not for you. You will need to wait or find another solution.

**Comment by Navid200** [comment]
An alternative approach would be to install the Libre app and use companion app to get readings from it into xDrip.
I know it will not be the same.  But, the advantage is it will not be interrupted by the manufacturer.

**Reply by Navid200** [reply]
Or, is that not an option?
If not, would there be an interest in adding it?

**Reply by Havanna22** [reply]
Does this companion app exist?

**Reply by Navid200** [reply]
The companion app would be the Libre linkup app.
You will need to have two apps installed.  One would be the official LibreLinkup app that you would use to follow.
And you will need to install xDrip.
If everything works after we add the app to the list, you would get readings from the Linkup app to xDrip as long as the Linkup app remains installed and operational.  It should also show notifications one for each reading including the glucose reading.

**Reply by Havanna22** [reply]
Aha! I think most users of this setup already have the LibreLinkUp app installed already in addition to XDrip+ to be able to check that it's working. I have that and see no problem with it

**Reply by Havanna22** [reply]
Meaning I have Libre3 app + LibreLinkUp + xDrip+ on same phone.

**Reply by Havanna22** [reply]
And it's not working, as of several hours, as many are reporting on FB and Reddit. Any fix would be much appreciated considering for example the awful alarm function in Libre3 app.

**Reply by Navid200** [reply]
Do you currently have readings in the librelinkup app?
I know you don't have readings in xDrip.
But, if you have readings in the librelinkup app, does it generate notifications for its readings?

**Reply by PhilTU** [reply]
> Do you currently have readings in the librelinkup app? I know you don't have readings in xDrip. But, if you have readings in the librelinkup app, does it generate notifications for its readings?

As far as I can tell (see FB) there is a notification generated if the master phone scans a Libre 2 sensor for data, or if there is an alarm, but not if the master phone is receiving data from the sensor by Bluetooth.

**Reply by Navid200** [reply]
If the master phone using the official app generates notifications, we ca add the master phone's app to the companion list.  Then,  the user can install xDrip on the master phone and get readings in it.
Then, xDrip on the master phone can be set up to be master either through xDrip Sync or Nightscout so that followers can get readings from xDrip.

This to me a solution that is unlikely to be rejected by the manufacturer because we will still be using their app.

I just don't understand everything you are describing.
Can this be used by everyone who is experiencing the problem, or only a portion of the users can do this?

**Reply by PhilTU** [reply]
> I just don't understand everything you are describing.
> Can this be used by everyone who is experiencing the problem, or only a portion of the users can do this?

there are multiple use cases, some without merit in my opinion.

Libre 3 sensors always send data by Bluetooth. No official app (master or follower) generates a notification every minute from the evidence so far.

Libre 2 sensors that are or need to be scanned by NFC *do* generate a notification in the follower LibreLinkUp app, as do alarms. These people may be able to benefit from companion app, but are probably in the minority.

Libre 2 sensors sending data by Bluetooth appear to be the same as Libre 3 - no notifications for glucose value.

Some users are "following themselves" with xDrip+ collecting data sent from offical app on same phone via cloud. Some of these are for incompatible sensors like the Libre 3 that won't connect directly.

Other users are following children or family members using xDrip+ that are remote / master is a different phone.

I hope that's clear.

**Reply by Navid200** [reply]
Yes, it is.  Thanks

**Comment by Filipesvs** [comment]
We are having the same issue. 
We have a "home based" mobile phone that we use as a "server" that collects glucose data from Librelinkup, sends to xdrip+, wich then uploads to nightscout, all on that phone.
And then, from nightscout, we as parents receive the data (from our daughter) on our mobile phones (xdrip)

Since the same time as the above reports, xdrip hasn´t been able to receive the data from librelinkup (librelinkup receives correctly the data on that phone).

Now we can´t upload the data to Nightscout, because this was done by xdrip+ on that "server" mobile phone.

Any help on how to solve this issue would be appreciated.

P.S: I installed Gluroo on the "server mobile phone" and Gluroo collects the data from librelinkup, but I can´t figure out how to upload the data from Gluroo to Nightscout, and was avoiding to use one more app on that phone.

**Reply by duncanScalion** [reply]
reached same conclussion ;)

 https://github.com/NightscoutFoundation/xDrip/discussions/3794#discussioncomment-11530170

maybe discussions should be merged

**Comment by pachi81** [comment]
### Workaround

Hi until the problem is fixed, I can provide a workaround:

1. Install GlucoDataHandler from Playstore or [GitHub](https://github.com/pachi81/GlucoDataHandler/releases/tag/1.2.1)
2. Setup LibreLinkUp and wait for data present
3. Go to Settings - Transfer values and activate "Send patched libre broadcast" and set xDrip+ as receiver
4. In xDrip select "Libre (patched app)" as source

**Reply by pachi81** [reply]
@Navid200: here you will find my fix (thanks @gui-dos):
https://github.com/pachi81/GlucoDataHandler/commit/16fc7639be4fc0b12b471b23c341438c037caddd#diff-6f48822c5d93a01a05ee987b6f652bdd4813a3b4c5be6495f379769498aa45a1

#### Changes to header:
version = 4.12.0
#### New to header:
Account-Id = userID as SHA256

Important: always set "Account-Id" even there is no User ID available. Without User ID it must be empty.

**Reply by Havanna22** [reply]
> ### Workaround
> Hi until the problem is fixed, I can provide a workaround:
> 
> 1. Install GlucoDataHandler from Playstore or [GitHub](https://github.com/pachi81/GlucoDataHandler/releases/tag/1.2.1)
> 2. Setup LibreLinkUp and wait for data present
> 3. Go to Settings - Transfer values and activate "Send patched libre broadcast" and set xDrip+ as receiver
> 4. In xDrip select "Libre (patched app)" as source

Thanks working 🤗🤗🤗🤗🤗

**Reply by Navid200** [reply]
@pachi81 I don't think changing xDrip knowing that the manufacturer will disapprove and again change their setup in near future is a solution.
I am looking for a solution that can persist.

**Reply by Filipesvs** [reply]
> ### Workaround
> Hi until the problem is fixed, I can provide a workaround:
> 
> 1. Install GlucoDataHandler from Playstore or [GitHub](https://github.com/pachi81/GlucoDataHandler/releases/tag/1.2.1)
> 2. Setup LibreLinkUp and wait for data present
> 3. Go to Settings - Transfer values and activate "Send patched libre broadcast" and set xDrip+ as receiver
> 4. In xDrip select "Libre (patched app)" as source

Thanks :-) Will try that later as soon as I arrive home, and will give feedback here

**Reply by pachi81** [reply]
> @pachi81 I don't think changing xDrip knowing that the manufacturer will disapprove and again change their setup in near future is a solution. I am looking for a solution that can persist.

@Navid200 Yes, that would be great, but as there is no notification or anything else, a persist solution is either patch the Libre App or use Juggluco instead. 
LibreLinkUp is also for me really annoying, as there are already more changes:
- new API api.libreview.ru (instead of the regions) 
- Error "2 - notAuthenticated" is the new error for nearly anything, like too many requests, wrong API server, new password policy and what ever...

So I think, there is no persist solution as to check the bundle.js of LibreLinkUp at each new release... It is terrible!

**Reply by PetBom** [reply]
The workaround works for me as well.

**Reply by PhilTU** [reply]
> So I think, there is no persist solution as to check the bundle.js of LibreLinkUp at each new release... It is terrible!

perhaps the "Web Follower" should be withdrawn ?

**Reply by Filipesvs** [reply]
> > ### Workaround
> > Hi until the problem is fixed, I can provide a workaround:
> > 
> > 1. Install GlucoDataHandler from Playstore or [GitHub](https://github.com/pachi81/GlucoDataHandler/releases/tag/1.2.1)
> > 2. Setup LibreLinkUp and wait for data present
> > 3. Go to Settings - Transfer values and activate "Send patched libre broadcast" and set xDrip+ as receiver
> > 4. In xDrip select "Libre (patched app)" as source
> 
> Thanks :-) Will try that later as soon as I arrive home, and will give feedback here

It worked, receiving values again on xdrip from GlucoDataHandler. And then updating the values from xDrip to Nighscout.

Thank you for your help

**Reply by Filipesvs** [reply]
Hello,

Giving an update, using Glucodatahandler works, although sometimes of the day there is a delay on the values, like 3 or 4 times a day we can have a delay of 30 or more minutes without receiving values.
Which rarely happened when using xdrip as webfollower.

**Reply by Havanna22** [reply]
Same here, gaps in the data from GlucoDataHandler.

**Reply by syska** [reply]
Have you turned off all battery optimizations for the app?

**Reply by davidcharlat** [reply]
when my phone is on standby for a piece of time, xDrip stops receiving values from GlucoDataHandler.
I have to restart xDrip to get the values again.
this didn't happen when I wasn't using GlucoDataHandler

**Reply by pachi81** [reply]
> when my phone is on standby for a piece of time, xDrip stops receiving values from GlucoDataHandler. I have to restart xDrip to get the values again. this didn't happen when I wasn't using GlucoDataHandler

Please check, if you also disabled battery optimization and so on for GlucoDataHandler. Also make sure that the notification is available for GlucoDataHandler.

**Reply by davidcharlat** [reply]
> Please check, if you also disabled battery optimization and so on for GlucoDataHandler. Also make sure that the notification is available for GlucoDataHandler.

yes battery optimization is disabled for both, and all notification are available for GlucoDataHandler.

I think the problem comes fron xDrip which doesn't work properly with GlucoDataHandler as my phone is in standby  ;(

**Reply by syska** [reply]
> > Please check, if you also disabled battery optimization and so on for GlucoDataHandler. Also make sure that the notification is available for GlucoDataHandler.
> 
> yes battery optimization is disabled for both, and all notification are available for GlucoDataHandler.
> 
> I think the problem comes fron xDrip which doesn't work properly with GlucoDataHandler as my phone is in standby ;(

I'm running the same setup and have no issues.

I'm not sure what "standby" means on a phone, since the phone is always running. But if your phone doesn't update xDrip when you are sleeping and the phone in by your side or in the charger. I don't have that issue.

**Reply by pachi81** [reply]
> > Please check, if you also disabled battery optimization and so on for GlucoDataHandler. Also make sure that the notification is available for GlucoDataHandler.
> 
> yes battery optimization is disabled for both, and all notification are available for GlucoDataHandler.
> 
> I think the problem comes fron xDrip which doesn't work properly with GlucoDataHandler as my phone is in standby ;(

Have you disabled Play Protect?

**Reply by davidcharlat** [reply]
83 / 5 000
>Have you disabled Play Protect?

I didn't do it, but now I did and the problem is still there

**Reply by Havanna22** [reply]
I have turned off all battery optimization on all relevant apps but still only get 4 readings per hour in XDrip, consistently using GlucoDataHandler.

Oddly enough, LibreLinkUp has started to lose values too, never happened before. Seems like every time I check it to see if it's ok, it says No recent data. Then I wait a while and suddenly a new value appears.
Gluroo seems to be doing a bit better but generally, I get very few readings per hour.

Has Abbott changed something so that Libre3 doesn't upload values every five minutes anymore, only every 15 minutes? Or can some values be "lost" because I have LibreLinkUp, Gluroo and GlucoDataHandler on the same phone?
Or did Abbott mess up their own apps with the recent updates?

I don't understand what's going on.
Might try juggcluco again, didn't get it to work when I tried once before but the setup I have now is not good enough.

**Reply by syska** [reply]
This is very odd ... everything now points to your phone.

Do you know if your LibreLink actually uploads? Have you tried using another phone as the follower?

**Reply by pachi81** [reply]
@Havanna22 This is a "normal" problem with Libre 3 app. You have reinstall the Libre 3 app!
Please also stop or uninstall LibreLinkUp app, as Abbott blocks too many requests.

Or use Juggluco, this is really the best option 😉

**Reply by Havanna22** [reply]
Thank you, thank you, thank you, yes reinstalling helped. 
Abbott's SW is SO frustrating to have to use. I will try with juggluco as soon as I have time. Last time I tried, I couldn’t start the sensor, again, not with Juggluco nor with Libre 3 so lost a sensor but  that was long ago. Worth a new try!

**Reply by pachi81** [reply]
@Havanna22 Most important thing for switch to Juggluco: setup your LibreView account first and make sure, you get an account ID. And before scanning your sensor you have to stop Libre 3 app.

**Reply by Havanna22** [reply]
1. I installed Juggluco
2. Libre 3 app had lost connection with sensor (intermittent error I guess)
3. I scanned the sensor (actually to get Libre 3 going again) which then activated Juggluco and I got the correct sound from activating/reading
4. I got the  Account ID succedsfully in Juggluco via my Libreview account
5. I went to setting and clicked all broadcasting options I could find
6. But I get no values in Juggluco :-(

Is it because I didn't stop Libre 3 app first?
Any tips on how to get values i Juggluco?

**Reply by pachi81** [reply]
Yes, you have to stop Libre 3 app, as the sensor can only connect to one app and the other one blocks the connection. Even if you want to change back to Libre 3 app for correct error code or anything else, you need to stop Juggluco, first.

**Reply by Havanna22** [reply]
Scanned the sensor again and now I have values!
Now just need to figure out how to get them to Xdrip (I did select boradcast to Patched Libre, maybe get values soon) and directly to my watch.

**Reply by pachi81** [reply]
@Havanna22 Yes the patched libre broadcast should be used and in xDrip Libre (patched app) as source. You can also install Juggluco on Wear OS watch.

**Reply by davidcharlat** [reply]
now i use juggluco instead of librelink, librelinkup & xDrip+ , and everything works fine

**Comment by PhilTU** [comment]
Ok, should I mark this as closed without resolution, escalate to an "issue", recommend removal of "Web follower" from xDrip+ as it is no longer fit for purpose or ........

**Reply by Navid200** [reply]
Discussions don't need to be closed regardless of whatever.  They are meant to be like forums, like facebook groups,  a way for everyone to share knowledge and ask questions.  

No, please do not change to issue.  

I may create a new web page to list all the requests as well as all the wished functionalities.

Please look at this open issue:  https://github.com/NightscoutFoundation/xDrip/issues/266
That is clearly an xDrip issue.  There is a problem in the code that does not allow treatments to be uploaded to the second site.  
This is not because a manufacturer has changed authentication to not allow us to use their site.    This is clearly because we have not added enough intelligence to the xDrip code to be able to upload to more than one site.

The function you want restored went away not because of a syntax error in xDrip.  You know why it has gone away.  The same as you, I wish this functionality to come back.  I know there is a way to bring it back.  I am not sure that is the right course of action if it will be removed again in 6 months by the manufacturer for the same reason.
This is not an xDrip issue.

I admit that someone may come up with a genius idea similar to the companion app feature to address this without bothering the manufacturer.  Therefore, it helps to have a way to track this.
However, I prefer not to flood the issues platform with everything we wished we had in xDrip.
That is why I am thinking to add a new web page on my site for keeping track of all such requests that are not recorded as an open issue.

**Reply by Navid200** [reply]
I prefer the issues platform to contain only a small number of issues that we know we must fix.  
Then, I can look at them and decide which one I need to work on.
If I look and see 1000 open issues, I may just walk away and go skiing!  I am not sure how to explain this.
There is no incentive for any of us to work on xDrip other than our belief in doing good for everyone to serve everyone including ourselves.  Demoralizing the few of us who are working on it by opening issue after issue after issue is not going to motivate us to work any harder.

**Reply by PhilTU** [reply]
It is frustrating because it seems to be a catch 22 where we know the functionality is broken (and why, and how to fix it) but we retain it without fixing it. So now there is a hardware data source that doesn't work. I understand your position and thanks for the explanation. It's one of those issues that arises in many open source type projects of management, resources, priorities etc to which I have no answer.

**Reply by Navid200** [reply]
How high can you set the threshold in the librelinkup app for the low alert?
Can you set it to 400mg/dL?
If you can and you do that, will you get a notification for every reading?

**Reply by Havanna22** [reply]
> How high can you set the threshold in the librelinkup app for the low alert? Can you set it to 400mg/dL? If you can and you do that, will you get a notification for every reading?

Max is 5.6 mmol/L

**Reply by Navid200** [reply]
And How low can you set the threshold for the high value alert?

**Reply by Havanna22** [reply]
> And How low can you set the threshold for the high value alert?

6.7

**Reply by HoLi1969** [reply]
Hi and happy New Year. I just installed the new nightly and the 4.12 version problem is still not fixed. As a solution was provided when llu version changed in the past but no programmer is doing the change now (why?) ... can a professional explain how to edit the code and build the APK new? xDrip worked perfectly with the Web follower so it is hard to step out just because the version needs to be adapted. Thanks for any help.

**Reply by pachi81** [reply]
> Hi and happy New Year. I just installed the new nightly and the 4.12 version problem is still not fixed. As a solution was provided when llu version changed in the past but no programmer is doing the change now (why?) ... can a professional explain how to edit the code and build the APK new? xDrip worked perfectly with the Web follower so it is hard to step out just because the version needs to be adapted. Thanks for any help.

I already checked the code, but it is done in an external script, not sure how to access it.
Also it is not only the version, there is a new mandatory element for header, so you need to change the script and the code.

**Comment by Havanna22** [comment]
For my understanding, other apps like Gluroo can get the values from
LibreLinkUp, why can't XDrip? Is Gluroo updated more frequently?

Den fre 13 dec. 2024 16:12PhilTU ***@***.***> skrev:

**Reply by Filipesvs** [reply]
Good question, if Gluroo and Glucodatahandler  can get values from LibrelinkUp, if Xdrip could do the same, it would be perfect!
Now we are using Glucodatahandler to "intermediate" the values between LibrelinkUp and Xdrip, if we could eliminate the middlemen :-)

**Reply by duncanScalion** [reply]
Gluroo does send 1minute notifications (It can have permanent glucose value in real time in notification area). Can this be used for xdrip problem? @Navid200 

On the other hand: i have never found any app such powerful as xdrip with graph and customization of alarms. Other app graphs are still slow, ugly or no readable at all. that keeps me prefering xdrip as "frontend"

**Reply by PhilTU** [reply]
> For my understanding, other apps like Gluroo can get the values from LibreLinkUp, why can't XDrip? Is Gluroo updated more frequently?

They fixed it, some others have fixed it and some have not (G-watch wear?)

**Comment by datacompboy** [comment]
Can we somehow fix the "beonlabs" script, is its source available? 🤔

**Reply by datacompboy** [reply]
oh, https://community-script.beonlabs.net returns error... so it doesn't work at all?

**Comment by jamorham** [comment]
If someone wants to privately share their follow login details with me via email then I can take a look at it. jamorham at gmail

**Reply by PhilTU** [reply]
Sent

**Comment by mariajosepela** [comment]
Yo también puedo enviar datos a @jamorham , si los necesitas, sólo tienes que decírmelo

**Reply by PhilTU** [reply]
please send it as it will give opportunity to test on a different country server (mine is on DE)

**Comment by Havanna22** [comment]
Working now! Just a small issue on the Watch: I got a very loud low BG
alarm, without possibility to silence, on my watch, even though I have
disabled alarms i Juggluco completely and have no sound on alarms on watch
and have disabled notifications from Juggluco on the watch. Odd, so if
there is a workaround for this, happy for tips. But I have also now set the
volume on my watch to zero, in addition to the already configured setting
of Vibrate Only, and hope not to get anymore loud watch notifications. (I
use the XDrip alarms, they are the best)

Den ons 18 dec. 2024 kl 08:02 skrev pachi81 ***@***.***>:

**Comment by HoLi1969** [comment]
Thanks, is there any source where I can learn how to do it? What tools are needed?

Gesendet von Outlook für Android<https://aka.ms/AAb9ysg>
________________________________
